reana_server/status.py

```diff
@@ -602,33 +604,40 @@ def __init__(self):
     def get_health_status(percentage):
         """Calculate quota health status."""
         health = ClusterHealthStatus.healthy
-        if percentage >= 50:
-            if percentage >= 75:
-                health = ClusterHealthStatus.critical
-            else:
-                health = ClusterHealthStatus.warning
+        if percentage <= 50:
+            health = ClusterHealthStatus.warning
+        if percentage <= 25:
+            health = ClusterHealthStatus.critical
+
         return health.name
 
     @staticmethod
-    def get_percentage(amount, total):
-        """Get percentage value based on ``amount`` and ``total``."""
-        return round((amount / total if total else 0) * 100)
+    def get_percentage(used, total):
+        """Get usage percentage based on ``used`` and ``total``."""
+        percentage = 100 - round((used / total if total else 0) * 100)
+        return percentage if percentage > 0 else 0
+
+    @staticmethod
+    def get_available(used, total):
+        """Get available cluster resources based on ``used`` and ``total``."""
+        available = total - used
+        return available if available > 0 else 0
 
     def get_node_health(self):
         """Get cluster nodes health information."""
         nodes_status_obj = NodesStatus()
 
         total = len(nodes_status_obj.get_nodes())
         unschedulable = len(nodes_status_obj.get_unschedulable_nodes())
+        available = ClusterHealth.get_available(unschedulable, total)
         percentage = ClusterHealth.get_percentage(unschedulable, total)
 
         return {
-            "available": total - unschedulable,
+            "available": available,
             "unschedulable": unschedulable,
+            "total": total,
             "percentage": percentage,
-            "health": ClusterHealthStatus.critical.name
-            if unschedulable > 0
-            else ClusterHealthStatus.healthy.name,
+            "health": ClusterHealth.get_health_status(percentage),
```

Comment:
I think that, if there's a single node that's `unschedulable`, it should be considered a critical situation. Under normal conditions, there shouldn't be any node that is `unschedulable`. That's probably why we kept this logic a bit different.

Imagine a situation where we have 20 nodes and 3 are malfunctioning, thus `unschedulable`. The health would still be `healthy` but actually, the cluster would need rapid attention.

Reply:
Thanks for bringing this question. It's true that `unschedulable` node should be very alarming, but I thought for regular users it could send a bit too strong message about our cluster health and node availability. 
In production we have ~30 nodes. Here is how two different approaches looks like:
![Screenshot from 2021-12-07 14-36-10](https://user-images.githubusercontent.com/4990025/145049800-a2b2c8c4-45df-4670-8a47-bee922a13576.png)
![Screenshot from 2021-12-07 14-37-22](https://user-images.githubusercontent.com/4990025/145049816-cd3cb8ae-f866-41e0-8db4-2254fca44914.png)

I agree that it's critical, just curious if we really want to show it to the end users. cc @tiborsimko

Reply:
I agree that we can display the "node slots" in red inside the circle. The plot also provides a message for us, the cluster admins, to get up and fix the nodes that have problems. 

As for the overall health badge having the number of 97% being displayed in red, I see that it may be too harsh to end users. If we have a cluster of say 500 nodes, then it is very well possible that some will be out "frequently" due to HW failures etc. The proability increases as the node count increases. Yet we would show 97% in red colour even if there are 485 nodes ready to server any heavy user workloads. That would be unnecessarily too pessimistic. We should accept that some nodes may have failures "naturally" in such a large cluster.

I guess we could use some logic such as above 90% the health badge is green, between 70% and 90% there are various shades of orange, and below 70% are various shades of red. Kind of like a battery gauge. Just an example.

Reply:
OK, let's leave it as it is then.